### ISSUE TITLE
CachedImage with GIF causes Nursery to give a warning/error.

### ISSUE THREAD

**Issue by elertan** [luberda-molinet/FFImageLoading#724]
http://pa1.narvii.com/6558/1b541e9955dccc15813f5ddad4431a5ef7339dd0_00.gif
Whenever I load the image url above into a CachedImage, that is imbedded into a ViewCell (So multiple occurences of the CachedImage are visible on the page) the app gets into a blocking/frozen state, and the Output on visual studio logs a Nursery warning/error.

I have tried other images that work fine, and other gifs as well. However this particular gif causes an error.

**Comment by daniel-luberda** [comment]
Hi @elertan. What platform?

**Comment by elertan** [comment]
Hey @daniel-luberda I have tried this on iOS. I can try to see if I can reproduce the error on an android device.

**Comment by daniel-luberda** [comment]
I just tested and couldn't reproduce your issue. I'll need some more info or a repro. 

![simulator screen shot 8 sep 2017 17 16 29](https://user-images.githubusercontent.com/13569983/30218546-93005a44-94b9-11e7-9df7-6f834aa96989.png)

**Comment by elertan** [comment]
I'm currently using this listview.
`<ListView ItemsSource="{Binding Chats}"
                      RowHeight="60">
                <ListView.ItemTemplate>
                    <DataTemplate>
                        <ViewCell>
                            <Grid>
                                <Grid.ColumnDefinitions>
                                    <ColumnDefinition Width="75" />
                                    <ColumnDefinition Width="6*" />
                                    <ColumnDefinition Width="2*" />
                                </Grid.ColumnDefinitions>
                                <Grid.RowDefinitions>
                                    <RowDefinition Height="*" />
                                    <RowDefinition Height="*" />
                                </Grid.RowDefinitions>
                                
                                
                                <ffImageLoading:CachedImage Grid.RowSpan="2"
                                                            Margin="10,0,0,0"
                                                            VerticalOptions="Center"
                                                            HorizontalOptions="Center"
                                                            WidthRequest="50"
                                                            FadeAnimationEnabled="True"
                                                            Source="{Binding ChatIcon}">

                                    
                                    <ffImageLoading:CachedImage.Transformations>
                                        <transformations:CircleTransformation BorderHexColor="#000000"
                                                                              BorderSize="1" />
                                    </ffImageLoading:CachedImage.Transformations>
                                </ffImageLoading:CachedImage>

                                
                                <Label FontAttributes="Bold"
                                       FontSize="14"
                                       Margin="0,5,0,0"
                                       Text="{Binding ChatTitle}"
                                       VerticalOptions="End"
                                       Grid.Row="0"
                                       Grid.Column="1" />
                                
                                <Label FontSize="13"
                                       Text="{Binding LastMessageText}"
                                       VerticalOptions="Start"
                                       LineBreakMode="TailTruncation"
                                       TextColor="DimGray"
                                       Grid.Row="1"
                                       Grid.Column="1" />
                                
                                <Label FontSize="10"
                                       Text="{Binding Chat.LastMessage.CreatedTime, Converter={StaticResource DateTimeToNowRelatedStringConverter}}"
                                       Margin="0,5,10,0"
                                       VerticalOptions="End"
                                       HorizontalOptions="End"
                                       TextColor="DimGray"
                                       Grid.Row="0"
                                       Grid.Column="2" />
                                
                            </Grid>
                        </ViewCell>
                    </DataTemplate>
                </ListView.ItemTemplate>
            </ListView>`

And the Chats property I am binding on is a ObservableRangeCollection<ChatCommunityModel>
Which looks like this
`public class ChatCommunityModel : BaseModel
    {
        private readonly string _myUserId;
        private Chat _chat;
        private Community _community;

        public ChatCommunityModel(string myUserId)
        {
            _myUserId = myUserId;
        }

        public string LastMessageText
        {
            get
            {
                if (Chat.LastMessage != null)
                {
                    return Chat.LastMessage.Content;
                }
                return String.Empty;
            }
        }

        public string ChatTitle
        {
            get
            {
                if (!string.IsNullOrWhiteSpace(Chat.Title)) return Chat.Title;
                return Chat.Members.First(m => m.Uid != _myUserId).Nickname;
            }
        }

        public string ChatIcon
        {
            get
            {
                if (Chat.Icon != null) return Chat.Icon.ToString();
                return Chat.Members.FirstOrDefault(m => m.Uid != _myUserId)?.Icon?.ToString();
            }
        }

        public Chat Chat
        {
            get => _chat;
            set
            {
                _chat = value;
                OnPropertyChanged();
            }
        }

        public Community Community
        {
            get => _community;
            set
            {
                _community = value;
                OnPropertyChanged();
            }
        }
    }`

And on one of those images it occurs, maybe the transformation causes it?

**Comment by daniel-luberda** [comment]
It's hard to guess. Can you make a reproduction?

**Comment by elertan** [comment]
Here is a reproduction
![devenv_2017-09-08_18-12-37](https://user-images.githubusercontent.com/9092847/30220862-6cdf90b6-94c1-11e7-8f6c-00fef6511950.png)

Here is one without the CircleTransformation, which still has the issue.
![devenv_2017-09-08_18-14-39](https://user-images.githubusercontent.com/9092847/30220926-ac5106d0-94c1-11e7-88ba-57ec555f0ed2.png)

**Comment by daniel-luberda** [comment]
I rather meant something that I could run by myself & debug it on my own :) Code looks good to me.

**Comment by elertan** [comment]
I'll try making a project and see if I can reproduce the error. If it still causes the same error, I'll send you a link to it.

**Comment by elertan** [comment]
I created a new app that reproduces the issue.
Here is a link to the project [https://mega.nz/#!8wUQWbDL!r4lsbTTDThwXWDIkR5e2fb-WRHNqnNCFUjm9DNxMEMs](url)

**Comment by elertan** [comment]
Any new thoughts on this issue @daniel-luberda ? And thanks for helping out as always : )

**Comment by Drutol** [comment]
Hello, I'm encountering this issue too, in native Android project.

**Comment by daniel-luberda** [comment]
Could you check if it's better in the newest prerelease? There were some fixes / optimisations.

**Comment by Drutol** [comment]
Unfortunately not much of a difference :(
(pre-507)
```
09-16 20:29:36.726 D/Mono    (16024): GC_TAR_BRIDGE bridges 0 objects 0 opaque 0 colors 0 colors-bridged 0 colors-visible 79 xref 0 cache-hit 0 cache-semihit 0 cache-miss 0 setup 0.06ms tarjan 1.02ms scc-setup 0.10ms gather-xref 0.00ms xref-setup 0.01ms cleanup 0.00ms
09-16 20:29:36.726 D/Mono    (16024): GC_BRIDGE: Complete, was running for 0.10ms
09-16 20:29:36.727 D/Mono    (16024): GC_MINOR: (Nursery full) time 0.78ms, stw 2.23ms promoted 0K major size: 5376K in use: 3951K los size: 6144K in use: 2655K
```

**Comment by daniel-luberda** [comment]
Do you experience a dead lock or out of memory exception?

**Comment by Drutol** [comment]
It's deadlock

**Comment by daniel-luberda** [comment]
It should be fixed in the newest stable (2.2.12), please reopen if it still occurs.

**Comment by Drutol** [comment]
Yeah it looks like it's gone! Thanks for your hard work! :)
